src/AsmResolver.Symbols.Pdb/Records/CodeViewSymbol.cs

```diff
@@ -37,4 +38,19 @@ public static CodeViewSymbol FromReader(PdbReaderContext context, ref BinaryStre
             _ => new UnknownSymbol(type, dataReader.ReadToEnd())
         };
     }
+
+    private protected T? GetLeafRecord<T>(PdbReaderContext context, uint typeIndex) where T : CodeViewLeaf
+    {
+        return context.ParentImage.TryGetLeafRecord(typeIndex, out var leaf)
+            ? leaf switch
+            {
+                T t => t,
+                UnknownCodeViewLeaf unknownLeaf => context.Parameters.ErrorListener.BadImageAndReturn<T>(
+                    $"{GetType().Name} contains an unknown leaf ({unknownLeaf.LeafKind}) at index {typeIndex:X8}."),
+                _ => context.Parameters.ErrorListener.BadImageAndReturn<T>(
+                    $"{GetType().Name} contains an incorrect leaf type ({leaf.GetType().Name}) at index {typeIndex:X8}.")
```

Comment:
To me the error message is a bit confusing now, as it is unclear what `typeIndex` is referring to now (the referenced leaf or the index of the leaf that is referencing the other leaf?).

Would it make more sense to rephrase this error as `"{GetType().Name} {TypeIndex:X8} references a leaf at {typeIndex:X8} that is of an unexpected type ({leaf.GetType().Name})."`?

Reply:
Also, we could probably avoid the `GetType().Name` and replace it with `LeafKind` instead.

Reply:
I standardized the wording based on your suggestion.